simapp/v2/app_di.go

```diff
@@ -51,10 +53,34 @@ func init() {
 	}
 }
 
+var (
+	singletonScopedStoreBuilder root.Builder
+	singletonScopedOnce         sync.Once
+)
+
+func ProvideSingletonScopedStoreBuilder() root.Builder {
```

Comment:
A singleton scoped Provider is required due to the pattern of double injections, once for client and once for a server.  An alternative design would be to add a StoreBuilder parameter to `AppCreator` but this seemed like a lighter lift and honestly not so bad since it's quite explicit.

Reply:
This provider will need to be copy/paste by all chain dev, can we host it somewhere else?

Reply:
yea can move to runtime and reference here, I'm hesitant to make into `runtime.init()` funcs just because of the singleton scope.. I'm hoping we uncover a way to remove the double injection at some point.